### DISCUSSION TITLE
Make '==' have a default behavior similar to IEqualityComparer<T>.Default ?

### DISCUSSION THREAD

**Discussion by YT-GameMaker** [dotnet/csharplang#7645] [Language Ideas]
I think `==` should have a default behavior similar to `IEqualityComparer<T>.Default`, rather than having to be manually implemented. Since the existing code that did not manually define `==` never use `==` for non-default behavior, so perhaps this situation can be changed without affecting existing code. This way, there's no need to restrict the use of `==` to generic types that implement the `==` interface.

**Comment by HaloFour** [comment]
That ship sailed 22 years ago.

**Comment by CyrusNajmabadi** [comment]
Closing out. This is one of many duplicate issues from the same user. Please do not keep spamming us for the same request.
